File: mini-core/src/main/java/com/mini/core/jdbc/model/Paging.java
package com.mini.core.jdbc.model;

import java.io.Serializable;
import java.util.List;

import static java.lang.Math.max;

public final class Paging<T> implements Serializable {
	private static final long serialVersionUID = 1L;
	private int page, limit, total;
	private List<T> rows;
	
	/**
	 * 指定页码数和每页条数初始化
	 * @param page  页码数
	 * @param limit 每页条数
	 */
	public Paging(int page, int limit) {
		this.limit = limit;
		this.page = page;
	}
	
	/**
	 * 设置总条数
	 * @param total 总条数
	 * @return 当前对象
	 */
	public final Paging setTotal(int total) {
		this.total = total;
		return this;
	}
	
	/**
	 * 获取页码数
	 * @return 页码数
	 */
	public final int getPage() {
		return max(1, page);
	}
	
	/**
	 * 获取每页条数
	 * @return 每页条数
	 */
	public final int getLimit() {
		return max(1, limit);
	}
	
	/**
	 * 获取总条数
	 * @return 总条数
	 */
	public final int getTotal() {
		return max(0, total);
	}
	
	/**
	 * 获取跳过的条数
	 * @return 跳过的条数
	 */
	public final int getStart() {
		int p = getPage() - 1;
		return p * limit;
	}
	
	/**
	 * 设置分页数据
	 * @param rows 分页数据
	 */
	public final void setRows(List<T> rows) {
		this.rows = rows;
	}
	
	/**
	 * 获取分页数据
	 * @return 数据列表
	 */
	public final List<T> getRows() {
		return this.rows;
	}
}

File: mini-jdbc/src/main/kotlin/com/mini/core/jdbc/model/Paging.kt
@file:Suppress("unused")

package com.mini.core.jdbc.model

import java.io.Serializable
import kotlin.math.max

/**
 * 分页控件
 *
 * @author xchao
 */
class Paging<T> private constructor() : Serializable {
    // 页码数
    var page: Int = 1
        private set(value) {
            field = max(1, value)
        }

    // 每页条数
    var limit: Int = 20
        private set(value) {
            field = max(1, value)
        }

    // 数据总条数
    var total: Int = 0
        set(value) {
            field = max(0, value)
        }

    // 数据库查询的起始位置
    var start: Int = 0
        get() = (page - 1) * limit
        private set

    // 查询数据结果
    var rows: List<T>? = null

    // 初始化构造方法
    constructor(page: Int = 1, limit: Int = 20) : this() {
        this.limit = limit
        this.page = page
    }

}